File: JS/DuasNotas.test.js
/**
 * @version JAVASCRIPT
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
const TEST = require('tape')
const OBJ = require('./DuasNotas.js')

TEST('Starting DuasNotas test...', (t) => {
    //Test 1
    t.assert(OBJ.hasTwoNotesChange(11, 23) === "possible", "Expect: possible")
    //Test 2
    t.assert(OBJ.hasTwoNotesChange(500, 650) === "possible", "Expect: possible")
    //Test 3
    t.assert(OBJ.hasTwoNotesChange(100, 600) === "impossible", "Expect: impossible")
    //Test 4
    t.assert(OBJ.hasTwoNotesChange(9948, 9963) === "possible", "Expect: possible")
    //Test 5
    t.assert(OBJ.hasTwoNotesChange(1, 2) === "impossible", "Expect: impossible")
    //Test 6
    t.assert(OBJ.hasTwoNotesChange(2, 4) === "impossible", "Expect: impossible")

    t.end()
})

File: GO/DuasNotas_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import "testing"

func TestDuasNotas(t *testing.T) {
	tables := []struct {
		x    int64
		y    int64
		want string
	}{
		{11, 23, "possible"},
		{500, 650, "possible"},
		{100, 600, "impossible"},
		{9948, 9963, "possible"},
		{1, 2, "impossible"},
		{2, 4, "impossible"},
	}

	for _, table := range tables {
		got := hasTwoNotesChange(table.x, table.y)

		if got != table.want {
			t.Errorf("Waiting for this %s but the return was this: %s", table.want, got)
		}
	}
}
